src/probabilities_estimators/histograms/rectangular_binning.jl

```diff
@@ -15,35 +18,113 @@ Binning instructions are deduced from the type of `ϵ` as follows:
     intervals that cover all data.
 4. `ϵ::Vector{Float64}` divides the i-th coordinate axis into intervals of fixed size
     `ϵ[i]`, starting from the axis minima until the data is completely covered by boxes.
+
+!!! note "Be aware when mixing methods"
+    If the range of the data along some dimension is `[r1, r2]`, then
+    `RectangularBinning(N)` results in a bin edge length of
+    `nextfloat(r2 - r1) / N)`. `nextfloat` is used to ensure data are completely
+    covered. Keep this in mind if you're switching between the `ϵ::Int` and `ϵ::Float64`
+    ways of constructing grids.
 """
 struct RectangularBinning{E} <: AbstractBinning
     ϵ::E
 end
 
+const ValidFixedBinInputs = Union{Number, NTuple}
+
 """
-    RectangularBinMapping <: Encoding
+    FixedRectangularBinning <: AbstractBinning
+    FixedRectangularBinning(ϵmin::E, ϵmax::E, N::Int) where E
+
+Rectangular box partition of state space where the extent of the grid is explicitly
+specified by `ϵmin` and `emax`, and along each dimension, the grid is subdivided into `N`
+subintervals.
+
+Binning instructions are deduced from the types of `ϵmin`/`emax` as follows:
+
+1. `E::Float64` sets the grid range along all dimensions to to `[ϵmin, ϵmax]`.
+2. `E::NTuple{D, Float64}` sets ranges along each dimension individually, i.e.
+    `[ϵmin[i], ϵmax[i]]` is the range along the `i`-th dimension.
```

Comment:
This is incorrect syntax. Notice that `E` is already the type of `emin` So `E` will never be `Float64`, as `E` is a data type. What you want to write is `E <: Real` and `E <: Tuple`.